.github/workflows/bumpdevversion.yml

```diff
@@ -0,0 +1,115 @@
+          export OLD_VERSION_STRING=$(yq read recipe/meta.yaml package.version)
+          echo ::set-output name=OLD_VERSION_STRING::$OLD_VERSION_STRING
+
+          ## Step 2b. Build new version string (e.g. 6.2.0.dev1+hijklmn)
+
+          # Step 2b i. Get current tag of latest dev build (e.g. 6.2.0)
+          export TAG=$(echo $OLD_VERSION_STRING | grep -o -P '.*(?=.dev)')
+
+          # Step 2b ii. Add 1 to the previous dev version number (e.g. 6.2.0.dev0+abcdefg would output 1, i.e. for dev1)
+          export DEV_VERSION_NUM=`expr 1 + $(yq read recipe/meta.yaml package.version | grep -o -P '(?<=dev).*(?=\+)')`
+
+          # Step 2b iii. Get shortened new sha hash value (e.g. hijklmn)
+          export SHORT_HASH=$(echo $NEW_SHA_HASH | cut -c 1-7)
+
+          # Step 3d. Finally build the new version string (e.g. 6.2.0.dev1+hijklmn)
+          echo ::set-output name=NEW_VERSION_STRING::$TAG.dev$DEV_VERSION_NUM+$SHORT_HASH
+
+      # Change development package version and source git revision
+      - name: Update package version and git revision
+        run: |
+          # Update the recipe/meta.yaml file with new version info
+          yq write --inplace recipe/meta.yaml package.version ${{ steps.version_info.outputs.NEW_VERSION_STRING }}
+          yq write --inplace recipe/meta.yaml source.git_rev ${{ steps.version_info.outputs.NEW_SHA_HASH }}
+
+          # Fix incorrectly formatted {{ compiler('c') }}
+          yq write --inplace recipe/meta.yaml requirements.build[1] "{{ compiler('c') }}"
+          sed --in-place "s/'{{ compiler(''c'') }}'/{{ compiler('c') }}/g" recipe/meta.yaml
+
+          # Ensure <two spaces>#<one space>[<expression>] form
+          sed --in-place "s/\ \#\ \[/  # [/g" recipe/meta.yaml
+
+      # Take a look at recipe/meta.yaml contents
+      - name: Check meta.yaml contents
+        run: cat recipe/meta.yaml
+
+      # Open a Pull Request from personal fork to the upstream repo
+      # The GMT_FEEDSTOCK_TOKEN personal access token secret should have
+      # a scope of "repo: public_repo" for access to public repositories
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3.4.0
+        with:
+          title: "Bump GMT from ${{ steps.version_info.outputs.OLD_VERSION_STRING }} to ${{ steps.version_info.outputs.NEW_VERSION_STRING }}"
+          commit-message: "Bumps GMT dev version from ${{ steps.version_info.outputs.OLD_VERSION_STRING }} to ${{ steps.version_info.outputs.NEW_VERSION_STRING }}"
+          author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
+          # reviewers: "weiji14"  # TODO change to maintainer on conda-forge/gmt-feedstock
+          base: devel
+          branch: update/${{ steps.version_info.outputs.NEW_VERSION_STRING }}
+          delete-branch: true
+          token: ${{ secrets.GMT_FEEDSTOCK_TOKEN }}
```

Comment:
Why not use the built in token for the `GenericMappingTools` fork? That would be a lot safer when combined with my comment above to only run this from the fork repo. In this case, you could just replace this with `secrets.GITHUB_TOKEN`. Using your personal token can be dangerous because it gives access to act as you on any repo. This token is specific for pushing to the particular repo running the action so potentially a lot less dangerous.

Reply:
Hopefully this would work with this action in particular. It might not since I'm not sure organizations can submit PRs but it's worth a try. Otherwise, creating a dummy GMTbot account and using its token is a lot safer. We can share the login details for that account with the dev team so that it doesn't depend on a single person.

Reply:
> Hopefully this would work with this action in particular. It might not since I'm not sure organizations can submit PRs but it's worth a try.

Yes, we might just have to try running this Github Action on `GenericMappingTools/gmt-feedstock` using `secrets.GITHUB_TOKEN` and see if it works.

> Otherwise, creating a dummy GMTbot account and using its token is a lot safer. We can share the login details for that account with the dev team so that it doesn't depend on a single person.

Sounds like a plan, but let's test to see if `secrets.GITHUB_TOKEN` works first before resorting to this. Oh, and in case you missed it, the pangeo folks did something similar with a pangeo-bot at https://github.com/conda-forge/pangeo-dask-feedstock/pull/8.

Reply:
Yep, sounds like a plan. I hadn't seen that on Pangeo thanks for sharing 🙂

Reply:
Ok, so I've tested using `secrets.GITHUB_TOKEN` on the `GenericMappingTool/gmt-feedstock` branch, The action manages to run up to creating a branch 'update/6.2.0.dev2+1e9d2fd', but fails when creating the PR to `conda-forge/gmt-feedstock` with the following error message:

```
Error: Resource not accessible by integration
Restore persisted git credentials
  /usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  http.https://github.com/.extraheader
  /usr/bin/git config --local --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  http.https://github.com/.extraheader AUTHORIZATION: basic ***
  /usr/bin/git config --local --unset http.https://github.com/.extraheader ^AUTHORIZATION:
  Unset config key 'http.https://github.com/.extraheader'
  /usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
  Persisted git credentials restored
```

Full logs are at https://github.com/GenericMappingTools/gmt-feedstock/runs/1225184192?check_suite_focus=true#step:7:440. I reckon we're close :slightly_smiling_face: Is there a way to set an organization-specific personal access token? Or maybe we can go the bot way.